### DISCUSSION TITLE
THT Command Line?

### DISCUSSION THREAD

**Discussion by aoirthoir** [joelesko/tht#20] [General]
I noticed in the documentation or somewhere on the GitHub that there was thinking about having THT run from the command line. what I'm curious about is if that's been implemented where I could use THT as a scripting language from the command line rather than PHP or Lua or bash for some projects that I want to run.

**Comment by joelesko** [comment]
It hasn't been implemented yet.  I'm open to having someone take a crack at it.  

I was thinking of having it happen in two phases. 

1) Allow CL scripts to be run in a THT app's `scripts` folder. For doing things like utilities and cron jobs. 

2) Run a tht script from anywhere. The interim php files would have to be compiled to some global temp folder.  Pathing would also take some time to get right, since right now it bases all of the paths off of the app root.

**Comment by aoirthoir** [comment]
I am just getting back into programming from a ten year break (I was trucking). And in addition to that I've not done PHP OOP (or any). But once i understand what you are doing I would be willing to figure it out. I would suggest the regular THT has priority tho until it reaches version one to your satisfaction.

Command line is interesting especially with what you have done with tht thus far, but its far more interesting and useful for most folks i would guess to first have web interface totally done.

I have studied a lot of the manual and really like it and i took a lot of notes. I will review again and validate my thoughts before posting and i want to test it all as well to make sure i really understand what you are doing.

I did look at the code that produces tht and its quite extensive and will take me some time to figure out. If you have any notes that explain the code i could go from there, if not I will just peer through as I am able.

I REALLY want this tht to become a major thing. When I have a better handle on it i am going to create some video tutorials...basically my opinion is tht is how PHP should have been done in the first place.  I am not terribly keen on OOP but the way you have done it, I actually could become keen on that.

Thanks
